src/ir/comp.rs

```diff
@@ -234,12 +261,24 @@ impl Component {
         self.expr_params_acc(expr, &mut acc);
         acc
     }
+
     /// Parameters mentioned within an expression
     pub fn prop_params(&self, prop: PropIdx) -> Vec<ParamIdx> {
         let mut acc = Vec::new();
         self.prop_params_acc(prop, &mut acc);
         acc
     }
+
+    /// Unannotated Ports in the component
+    pub fn unannotated_ports(&self) -> Vec<(&ast::Id, u64)> {
+        self.info
+            .iter()
+            .filter_map(|(_, info)| match info {
+                Info::UnannotatedPort { name, width } => Some((name, *width)),
+                _ => None,
+            })
+            .collect()
+    }
```

Comment:
This will be a costly operation since `self.info` can get quite large.

Reply:
Should we directly put unannotated ports into the component instead?